main.nf

```diff
@@ -180,29 +200,113 @@ process get_software_versions {
 }
 
 /*
- * STEP 0: Extract reads mapping to specific chromosome(s)
+ * If index_files not provided as input
  */
-if (params.chr){
-  process extractReadsMappingToChromosome{
-    tag "${name}.${chr_list_joined}"
-    label 'process_medium'
+
+if(!params.index_files){
+	process IdxBAI {	
+		tag "$name"
+		label 'process_medium' 
+
+		input:
+		set val(name), file(bam) from bam_files_index		
+	
+		output:
+		file "${name}.bai" into ch_index_files 
+		file "${name}.bai" into ch_chr_index_files
+	
+		
+		script:	
+		"""		
+		samtools index $bam > ${name}.bai	
+		"""		
+	}
+	
+	// Extract reads mapping to specific chromosome(s)
+	if (params.chr){
+	  process extractReadsMappingToChromosome{
+	    tag "${name}.${chr_list_joined}"
+	    label 'process_medium'
   
-    input:
-    set val(name), file(bam) from bam_chr 
+	    input:
+	    set val(name), file(bam) from bam_chr 
+		  set val(name), file(bai) from ch_chr_index_files
```

Comment:
```suggestion
	set val(name_bai), file(bai) from ch_chr_index_files
```

Reply:
Here it is best to give different names to the two variables, even if they have the same content.